src/v2/guide/accessibility.md

```diff
@@ -0,0 +1,301 @@
+    <div class="toggle-button__handle"></div>
+  </div>
+  
+  <pre>{ checked: {{ checked }} }</pre>
+</div>
+<script>
+var example1 = new Vue({
+  el: '#example-2',
+  data: {
+    checked: false,
+  },
+})
+</script>
+{% endraw %}
+
+The `sr-only` class means that the checkbox is visible only to screen readers, and `aria-hidden="true"` on the toggle button means that it is hidden to screen readers. Screen readers read out the checkbox, and your sighted users see the toggle button as intended—it works for both parties.
+
+#### ARIA attributes
+
+We saw an ARIA (short for [Accessible Rich Internet Applications][WAI-ARIA]) attribute in the previous example to hide the custom HTML from the screen reader. aria-hidden is one of a tonne of attributes that you can add to elements to tell assistive technology such as screen readers what things on the page are.
+
+Check out MDN for a longer read on ARIA: [An overview of accessible web applications and widgets][ARIA MDN].
+
+Additionally, the W3C keeps a detailed guide of usage and examples for ARIA: [Using Aria - W3C Working Draft][Using ARIA]. This also includes a handy list of [ARIA roles][ARIA Roles] and [ARIA states and properties (or attributes)][ARIA States and Properties].
+
+To make first example accessible using ARIA, we have to set the role to "checkbox" and use the `aria-checked` attribute:
+
+```html
+<div
+  role="checkbox"
+  tabindex="0"
+  :aria-checked="checked"
+  :class="['toggle-button', { 'toggle-button--checked': checked }]"
+  @click="checked = !checked">
+  <div class="toggle-button__handle"></div>
+</div>
+```
+
+{% raw %}
+<div id="example-3" class="demo">
+  <div
+    role="checkbox"
+    tabindex="0"
+    :aria-checked="checked"
+    :class="['toggle-button', { 'toggle-button--checked': checked }]"
+    @click="checked = !checked">
+    <div class="toggle-button__handle"></div>
+  </div>
+  
+  <pre>{ checked: {{ checked }} }</pre>
+</div>
+<script>
+var example1 = new Vue({
+  el: '#example-3',
+  data: {
+    checked: false,
+  },
+})
+</script>
+{% endraw %}
+
+Either method of making the custom input element accessible works fine and isn't too difficult to implement.
+
+### Dynamic content
+
+when content is updated, screen readers don't know it has been updated
+
+examples:
+
+- error messages on forms
+- new information (eg news, notifications) from server
+- requested information finished downloading
+
+you can use aria-live to let the user know that stuff has changed
+
+
+### Page navigation with client-side routing
+
+when you click a link in a website with server-side routing, the browser knows the page has changed. on a website with client-side routing, the browser doesn't know, so it isn't announced to the user - it just announces that the link has been clicked, not that navigation has occured or what is displayed has changed.
+
+the solution is to announce to the user when the route has changed using a hidden aria-live element.
+
+## Testing accessibility
```

Comment:
How about add a section to introduce some a11y tools or component libs with good a11y support in Vue ecosystem. I heard some of them:

* [VEUI](https://github.com/ecomfe/veui) - component lib
* [Element](https://github.com/ElemeFE/element) - component lib
* [eslint-plugin-vue-a11y](https://github.com/maranran/eslint-plugin-vue-a11y) - eslint plugin

And I personally just started a project called [Vue A11y Utils](https://github.com/jinjiang/vue-a11y-utils) to try supplying some a11y utilities for Vue. It's still wip, not finished. 😜

Thanks.

Reply:
I'd suggest only including libraries that pass an accessibility review. For instance, I wouldn't include Element in this list as a quick spot check some of the Element components and there is still a bit of work to do. The same spot check on VEUI reveals that best practices are mostly being followed, but I found VEUI still needs some help with contrast and focus indicators.